### PR TITLE
PropProof integration

### LINKED ISSUE #15: Proptest support within Kani Action.
Requested feature: PropProof integration (https://github.com/model-checking/kani/issues/1608) should work inside action.
Use case: PROST, et. al.

Test case: https://github.com/YoshikiTakashima/simplified-prost-rs-timestamp

### PR DESCRIPTION
**Pull request by YoshikiTakashima**
### Description of changes: 

This PR enables the PropProof feature in the kani action

### Resolved issues:

Resolves #15 

### Call-outs:

- The action is set up to run with PropProof enabled by default. ~~I think this would be better as (1) having explicitly disable a harness is more sound than forgetting to run it, and (2) this avoids customer-side flags to enable proptest, avoiding issues when the interface changes.~~
- As per below discussion, we will include an explicit flag to enable prop proof.

### Testing:

* How is this change tested? `tests/cargo-kani/proptest-lib`

* Is this a refactor change? No.

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

**Comment by YoshikiTakashima** [comment]
@adpaco-aws @celinval Would you mind bringing this up for discussion, on the deep dive maybe?

Also please approve the CI run above.

**Comment by adpaco-aws** [comment]
@YoshikiTakashima we didn't have time to discuss it this week. Please allow us to discuss in the first weeks of January - I'll get back to you as soon as I can.

**Comment by YoshikiTakashima** [comment]
@adpaco-aws. Thanks you for pushing on this topic. 

As of  https://github.com/model-checking/kani-github-action/pull/19/commits/403a0016547614d0373c7c95640fe7c5f96ec21f, the code should be ready to go once the discussion is concluded.

**Comment by adpaco-aws** [comment]
Hi @YoshikiTakashima , we discussed this and we'd prefer using a flag to enable the PropProof feature since it's unstable and we don't want to surprise our users.

**Comment by YoshikiTakashima** [comment]
Thanks. I’ll implement asap.

**Comment by YoshikiTakashima** [comment]
@adpaco-aws Explicit PropProof flag is ready to go as of 3f754e92156aef5f2e1afac3bca2e72587b79774

**Comment by YoshikiTakashima** [comment]
@adpaco-aws is there more issues that need to be resolved?

otherwise, do you have an eta for the review? I Would like to get it merged soon since we have potential customers who agreed to integrate with this method.

Thanks for all the work so far :)

File: action.yml [hunk 1]
```diff
@@ -22,6 +22,10 @@ inputs:
     description: 'Arguments to pass to kani.'
     required: false
     default: ''
+  enable-propproof:
+    description: 'Experimental: Allow Kani to verify proptest harnesses using the propproof feature'
```
Comment (adpaco-aws) [inline action.yml:26]:
```suggestion
    description: 'Experimental: Allow Kani to verify proptest harnesses using the PropProof feature'
```

File: tests/cargo-kani/proptest-lib/fails.expected [hunk 2]
```diff
@@ -0,0 +1,4 @@
+SUCCESS\
```
Comment (adpaco-aws) [inline tests/cargo-kani/proptest-lib/fails.expected:1]: We are only running `works_fine`, right? I don't understand why this file is being included.

File: tests/cargo-kani/proptest-lib/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        fn works_fine(even_num in any::<u32>().prop_map(|x| x << 1)) {
+            prop_assert_eq!(even_num % 2, 0, "even number");
+        }
+    }
+
+    #[cfg(not(kani))]
+    mod expected {
+        proptest! {
+            /// Overflow by check in x * 2. TODO: Need to implement
+            /// expected fail tests. tracking issue #5
```
Comment (adpaco-aws) [inline tests/cargo-kani/proptest-lib/src/lib.rs:20]: Please paste the full URL to the issue.

File: tests/cargo-kani/proptest-lib/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,26 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! basic proptest examples. First one fails but second does not.
```
Comment (adpaco-aws) [inline tests/cargo-kani/proptest-lib/src/lib.rs:4]: Isn't it the other way around?

File: tests/cargo-kani/proptest-lib/fails.expected (hunk 2 above)
Comment (YoshikiTakashima) [reply tests/cargo-kani/proptest-lib/fails.expected:1]: I don't think it has any use as of now. This file is just leftover from when I was experimenting.

File: tests/cargo-kani/proptest-lib/src/lib.rs (hunk 4 above)
Comment (YoshikiTakashima) [reply tests/cargo-kani/proptest-lib/src/lib.rs]: Fixed with d25e6d296c9bfb4c0e83171eeb5a6d75666d8b18

File: tests/cargo-kani/proptest-lib/fails.expected (hunk 2 above)
Comment (YoshikiTakashima) [reply tests/cargo-kani/proptest-lib/fails.expected]: Fixed with d25e6d296c9bfb4c0e83171eeb5a6d75666d8b18

File: tests/cargo-kani/proptest-lib/src/lib.rs (hunk 3 above)
Comment (YoshikiTakashima) [reply tests/cargo-kani/proptest-lib/src/lib.rs]: Fixed with d25e6d296c9bfb4c0e83171eeb5a6d75666d8b18

**Comment by YoshikiTakashima** [comment]
@adpaco-aws Comments are fixed. Please merge when you think its ready.

**Review by adpaco-aws — APPROVED** [review]

**Pull request merged**